FRDeviceBinding/FRDeviceBinding/Sources/DeviceBindingAuthenticators.swift

```diff
@@ -81,9 +88,14 @@ extension DeviceAuthenticator {
     /// - Parameter userId: user Id received from server
     /// - Parameter challenge: challenge received from server
     /// - Parameter expiration: experation Date of jws
+    /// - Parameter customClaims: A dictionary of custom claims to be added to the jws payload
     /// - Returns: compact serialized jws
     /// - Throws: `DeviceBindingStatus` if any error occurs while signing
-    public func sign(keyPair: KeyPair, kid: String, userId: String, challenge: String, expiration: Date) throws -> String {
+    public func sign(keyPair: KeyPair, kid: String, userId: String, challenge: String, expiration: Date, customClaims: [String: Any] = [:]) throws -> String {
+        guard customClaims.isEmpty || validateCustomClaims(customClaims) else {
```

Comment:
- Move the check earlier, under DeviceBindingCallback and DeviceSigningVerifierCallback
- Instead of check empty first, can we move the check empty inside validateCustomClaims? if the list is empty consider valid.
- Since the caller is provide invalid argument, Shoud we throw invalidargumentexception?

Reply:
@witrisna 

- Moved _validateCustomClaims_ call into  **DeviceBindingCallback** and **DeviceSigningVerifierCallback**
- Removed the empty check all together, _validateCustomClaims_ method returns true for empty claims already
- In swift we don' have compile time invalidargumentexception the way java has, and also exceptions are handled differently in swift, they are runtime. I could call _fatalError()_ which is compile time error, but not sure that's what we want.